### DISCUSSION TITLE
DDS Repo Catalogue

### DISCUSSION THREAD

**Discussion by oraqlle** [vector-of-bool/dds#120] [General]
I know you're busy with the beta release of DDS but I was wondering what you had planned to encourage library vendors to support DDS as a build system and add packages to their own repo or upload them to [repo-1.dds.pizza](https://repo-1.dds.pizza/). I would love to see (after the release) a growth in the accessible packages. 
Is this done with [dds ports](https://github.com/vector-of-bool/dds-ports) or by making forks of the repo.
If you would like help updating and preparing the catalogue for the dds repo, I can offer some of my time to help make that happen. I feel like this tool will be integral for future C++ development and I'd love to see it grow and help out if I could.

**Comment by david-bakin** [comment]
To this end I'm wondering if [`package-schema.json`](https://github.com/vector-of-bool/dds/blob/develop/res/package-schema.json) could be a bit less locked down, or have an extra property or two defined now (even if unused in the code).  I'm specifically thinking of a `description` property.  Possibly treated like a git "commit" message, i.e., if formatted as a single (short) line followed (optionally) by a blank line and then more text, to be "brief" and "full".

Idea there would be that ultimately someone could provide a tool to create a better top-level "index" page for a DDS repo.

**Reply by vector-of-bool** [reply]
The packaging format is being updated and will support such metadata soon. I'm attempting to make the package format dds-agnostic so that other package+build systems will be able to pull/push/build packages in the future without requiring all of dds to do so.

**Reply by oraqlle** [reply]
That sounds awesome. Especially for existing libraries and such that heavily rely on stuff like Cmake (if I'm understanding correctly).

**Reply by oraqlle** [reply]
And descriptions of the package would be a great feature.

**Comment by oraqlle** [comment]
And if when packaging a project for a repo, if that repo were a static site, built with tools like Hugo or Zola, it'd be cool if dds could pull the readme and add it to a documentation folder soe you could build a website that can display some information about packages. 
Hopefully this isn't overwhelming, just ideas that could attract more users of dds.

**Comment by vector-of-bool** [comment]
New packages are added via PR to the `dds-ports` repository. For most packages, they will automatically discover new releases by pulling the Git repository tags. It isn't 100% resilient, but is sufficient for many cases. Many libraries can be added with only a few lines, depending on the repository layout (Refer to `ports/simple_ports.py`). I am working on a refactor of much of this code in the `feature/crs-repo` branch, but a lot of the port declarations look mostly unchanged.

**Reply by oraqlle** [reply]
Awesome, keep up the awesome work. I know at least that this is a tool I've been looking for in C++ and I feel is sorely missing.

**Reply by david-bakin** [reply]
@vector-of-bool - can you explain a little more about how the automatic new release thing works?  Esp. _when_ it happens?  I.e., if I create a personal repo (say a GitLab repo published via GitLab Pages) - how do I get automatic new releases?

**Reply by vector-of-bool** [reply]
dds/bpt will pull the repository data every time it attempts to do dependency resolution (which happens on every build). It uses HTTP cache controls, timestamps, and ETags to minimize the frequency of needing to download the data.
